src/storage/data_table.cpp

```diff
@@ -184,29 +201,34 @@ TupleSlot DataTable::Insert(const common::ManagedPointer<transaction::Transactio
       common::SpinLatch::ScopedSpinLatch guard(&blocks_latch_);
       // insert block
       blocks_.push_back(new_block);
-      block = --blocks_.end();
+      block = new_block;
       break;
     }
 
-    if (accessor_.SetBlockBusyStatus(*block)) {
+    {
+      common::SpinLatch::ScopedSpinLatch guard(&blocks_latch_);
```

Comment:
Taking the latch here is concerning to me. Definitely look out for microbenchmark numbers on this one.

Reply:
You have to take the latch if you're going to do an index lookup on the vector (otherwise it could migrate underneath your dereference because of an insert.